operations/configure_cray_cli.md

```diff
@@ -21,50 +25,195 @@ The `cray` CLI only needs to be initialized once per user on a node.
+   the script with the individual initialization option and debug level logging enabled:
+   ```bash
+   ncn# ssh ncn-m003
+   ncn-m003# python3 /tmp/craycli_init.py --initnode --debug
+   ```
+
+   Now use the enhanced messages to determine what is wrong on this node.
+
+1. Keycloak user group membership
+
+   The keycloak user that is used to initialize and authorize the `cray` CLI must belong to
+   a group that has permissions to run the `cray` CLI. In a default install the group `craydev`
+   is set up in keycloak with the correct permissions and the temporary user created by this
+   process will be added as a member of that group. If there is a different group that has the
+   correct permissions the `--group` option can be used on the `--run` step to use that group
+   instead:
+
+   ```bash
+   ncn# python3 /usr/share/doc/csm/install/scripts/craycli_init.py --run --group my_grp
+   2021-12-21 15:50:47,814 - INFO    - Loading keycloak secrets.
+   2021-12-21 15:50:48,095 - INFO    - Created user 'craycli_tmp_user' in 'my_grp'
+   2021-12-21 15:50:52,714 - INFO    - Initiailizing nodes:
+   2021-12-21 15:50:52,714 - INFO    - ncn-m001: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-m002: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-m003: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-s001: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-s002: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-s003: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-w001: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-w002: Success
+   2021-12-21 15:50:52,714 - INFO    - ncn-w003: Success
+   ```
+
+1. Missing Python Modules
+
+   It is possible that some Python modules required for the script are missing on individual
+   nodes - particularly on the `PIT` node if that is still active. This script could run from
+   any of the NCN nodes, so if it fails on one node, copy it to any location on another node
+   and try to run it from there.
+
+   In the below example the script fails on 'ncn' due to the missing Python module 'oauthlib'
+   so it is copied to 'ncn-m002' and successfully runs from that node:
+   ```bash
+   ncn# python3 /usr/share/doc/csm/install/scripts/craycli_init.py --run
+   Traceback (most recent call last):
+     File "craycli_init.py", line 50, in <module>
+       import oauthlib.oauth2
+   ModuleNotFoundError: No module named 'oauthlib'
+   ncn# scp /usr/share/doc/csm/install/scripts/craycli_init.py ncn-m002:~/my_dir/
+   ncn# ssh ncn-m002
+   ncn-m002# cd my_dir
+   ncn-m002# python3 ./craycli_init.py --run
+   ```
+
+   At this point expect it to proceed as documented, but it will fail again on the node
+   originally attempted on due to the lack of critical Python modules on that node, but
+   may complete successfully on the rest of the nodes.
+
+   Alternatively the modules could be installed using 'pip' or 'pip3' if that is available on the node.
+
```

Comment:
It might be a good idea to add another small header here to differentiate the different sets of steps in this troubleshooting section.

Maybe...

### Troubleshoot Failed Initialization